### ISSUE TITLE
Can I save my checkpoints for the sentence transformers and resume training when interrupted?

### ISSUE THREAD

**Issue by SushaSureshh** [UKPLab/sentence-transformers#1008]
Hello All,

Is it possible to save the checkpoint let us say after 200 steps and then resume training from the latest checkpoint if the training is getting interrupted?

**Comment by mamoon115** [comment]
Hello, @SushaSureshh Did you get the solutions for that problem?

**Comment by nreimers** [comment]
Hi 
The fit method has several checkpoint parameters you can use for this:
https://www.sbert.net/docs/package_reference/SentenceTransformer.html#sentence_transformers.SentenceTransformer.fit

**Comment by SushaSureshh** [comment]
hello @mamoon115 I saw the code that @nreimers sent in the comments above. I wrote a custom code to save the `current_step` before the training was interrupted and resume training from the latest checkpoint if the `current_step != total_steps`.

**Comment by mmm898** [comment]
> hello @mamoon115 I saw the code that @nreimers sent in the comments above. I wrote a custom code to save the `current_step` before the training was interrupted and resume training from the latest checkpoint if the `current_step != total_steps`.

Hi, I'm having the same problem and i still can't solve it. Can you show me your code, I can't find my last checkpoint to resume training from it

**Comment by gonchik** [comment]
Hi @nreimers , 
+1 it will be nice to know. 
Cheers,
Gonchik

**Comment by shrijayan** [comment]
> Hello All,
> 
> Is it possible to save the checkpoint let us say after 200 steps and then resume training from the latest checkpoint if the training is getting interrupted?

model.fit(train_objectives = ,
          epochs = ,
          warmup_steps = ,
         checkpoint_save_steps = 256)
         
Use checkpoint_save_steps = Will save a checkpoint after so many steps

**Comment by david-waterworth** [comment]
@shrijayan but then how do you resume training from the last one?

**Comment by tomaarsen** [comment]
Resuming training is not really possible before v3.0, because the `model.fit` method doesn't save the optimizer/scheduler state, what step you were on, etc. All you can do is train with all data from scratch again, but using a different initial checkpoint.

Resuming your training will become possible in v3.0 (set to release this week) by calling `trainer.train()` with a path to your checkpoint. The v3 checkpoints *will* save the trainer state, allowing you to properly resume training as if you never took a break at all.

- Tom Aarsen

**Comment by david-waterworth** [comment]
Awesome thanks! That'll be really helpful because it's a requirement for training with AWS Spot instances (much cheaper)!

**Comment by tomaarsen** [comment]
Makes sense! Release should come really soon.
Also, it'll introduce multi-gpu training, that might also be helpful when looking to accelerate your training over AWS.

- Tom Aarsen
